### ISSUE TITLE
`import xxx as var` does not narrow type

### ISSUE THREAD

**Issue by spagh-eddie** [python/mypy#14957]
**Bug Report**

I would expect a variable that gets imported narrow the type.

**To Reproduce**

https://mypy-play.net/?mypy=latest&python=3.11&gist=11a4883cf29edd47fd1aff15302ead30

```python
x: str | None = None
if x is None:
    from os import sep as x
# revealed type is Optional[str]
print(x + "a")  # mypy false positive
```

**Expected Behavior**

No error. 

**Actual Behavior**

```console
% mypy test.py
main.py:5: error: Unsupported left operand type for + ("None")  [operator]
main.py:5: note: Left operand is of type "Optional[str]"
Found 1 error in 1 file (checked 1 source file)
```

```console
% python test.py
/a
```

**Your Environment**

- Mypy version used: 1.1.1
- Mypy command-line flags: none
- Python version used: 3.11

**Comment by AlexWaygood** [comment]
Duplicate of #12965

**Comment by spagh-eddie** [comment]
> Duplicate of #12965

nice! I didn't find it... I assume because searching "import" has many false positives
